util/src/main/scala/com/scalableminds/util/io/ZipIO.scala

```diff
@@ -100,22 +100,27 @@ object ZipIO {
 
     def isFileHidden(e: ZipEntry): Boolean = new File(e.getName).isHidden || e.getName.startsWith("__MACOSX")
 
-    def stripPathFrom(path: Path, string: String): Option[String] = {
-      if(path.toString.contains(string))
-        Some(path.toString.substring(0, path.toString.indexOf(string)))
-      else
-        None
+    def stripPathFrom(path: Path, string: String): Option[Path] = {
+      for (i <- 0 until path.getNameCount) {
```

Comment:
I would prefer this to be recursive, but I think it's still fine

Reply:
Sounds good I'll do it later